### PR TITLE
Merge atree inlining cadence v1.0

### PR DESCRIPTION
**Pull request by turbolent**
This is merging the atree inlining feature to `master` 🎉 🎉 🎉 

Amazing work @fxamacker 👏 👏 👏

### CODE REVIEW

File: fvm/evm/testutils/cadence.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ func EncodeArgs(argValues []cadence.Value) [][]byte {
 	return args
 }
 
+
```
Comment (SupunS) [inline fvm/evm/testutils/cadence.go:78]:
```suggestion
```

**Review by SupunS — APPROVED** [review]

File: fvm/evm/emulator/state/collection.go [hunk 2]
```diff
@@ -47,16 +47,22 @@ func NewCollectionProvider(
 // calling twice for the same collection might result in odd-behaviours
 // currently collection provider doesn't do any internal caching to protect against these cases
 func (cp *CollectionProvider) CollectionByID(collectionID []byte) (*Collection, error) {
-	storageID, err := atree.NewStorageIDFromRawBytes(collectionID)
+	slabID, err := atree.NewSlabIDFromRawBytes(collectionID)
 	if err != nil {
 		return nil, err
 	}
+
+	// TODO: expose SlabID.Address() in atree
```
Comment (turbolent) [inline fvm/evm/emulator/state/collection.go:55]: @fxamacker is this maybe available now?

File: fvm/evm/testutils/backend.go [hunk 3]
```diff
@@ -221,7 +221,7 @@ type TestValueStore struct {
 	GetValueFunc             func(owner, key []byte) ([]byte, error)
 	SetValueFunc             func(owner, key, value []byte) error
 	ValueExistsFunc          func(owner, key []byte) (bool, error)
-	AllocateStorageIndexFunc func(owner []byte) (atree.StorageIndex, error)
+	AllocateStorageIndexFunc func(owner []byte) (atree.SlabIndex, error)
```
Comment (turbolent) [inline fvm/evm/testutils/backend.go:224]: We should maybe change the remaining "storage index" naming to "slab index"

**Review by ramtinms — APPROVED** [review]
Nice work 👏

File: fvm/evm/testutils/backend.go (hunk 3 above)
Comment (turbolent) [reply fvm/evm/testutils/backend.go]: Improved this a bit in [8e4c7dc](https://github.com/onflow/flow-go/pull/6275/commits/8e4c7dcab0c76b1385483c2648da047a4b975133)

**Review by fxamacker — APPROVED** [review]
Nice!  Thanks for merging atree inlining feature branch with [atree v0.8.0-rc.5](https://github.com/onflow/atree/releases/tag/v0.8.0-rc.5) to master! :+1:

**Comment by turbolent** [comment]
Needs an Emulator update and I noticed `TestContractCleanupMigration2` is non-deterministic

File: fvm/evm/emulator/state/collection.go (hunk 2 above)
Comment (fxamacker) [reply fvm/evm/emulator/state/collection.go:55]: @turbolent Not yet, I just opened https://github.com/onflow/atree/issues/431 to add `SlabID.Address()`.

I can replace the workaround here after I add `SlabID.Address()` to atree.

**Comment by turbolent** [comment]
Opened https://github.com/onflow/flow-emulator/pull/721 and https://github.com/onflow/flow-go/pull/6276

**Pull request merged**